homeassistant/components/synology_dsm/__init__.py

```diff
@@ -463,13 +532,13 @@ async def async_update(self, now=None):
             _LOGGER.warning(
                 "async_update - connection error during update, fallback by reloading the entry"
             )
-            _LOGGER.debug("async_update - exception: %s", err)
+            _LOGGER.debug("SynoAPI.async_update() - exception: %s", err)
             await self._hass.config_entries.async_reload(self._entry.entry_id)
             return
         async_dispatcher_send(self._hass, self.signal_sensor_update)
 
 
-class SynologyDSMEntity(Entity):
+class SynologyDSMBaseEntity:
```

Comment:
```suggestion
class SynologyDSMBaseEntity(Entity):
```

Reply:
What is about this solution?
```diff
--- a/homeassistant/components/synology_dsm/__init__.py
+++ b/homeassistant/components/synology_dsm/__init__.py
@@ -21,6 +21,7 @@ from synology_dsm.exceptions import (
 )
 import voluptuous as vol
 
+from homeassistant.config import CONF_TEMPERATURE_UNIT
 from homeassistant.config_entries import SOURCE_IMPORT, ConfigEntry
 from homeassistant.const import (
     ATTR_ATTRIBUTION,
@@ -577,7 +578,7 @@ class SynologyDSMBaseEntity:
     def unit_of_measurement(self) -> str:
         """Return the unit the value is expressed in."""
         if self.entity_type in TEMP_SENSORS_KEYS:
-            return self.hass.config.units.temperature_unit
+            return CONF_TEMPERATURE_UNIT
         return self._unit
 
     @property
```

Reply:
`CONF_TEMPERATURE_UNIT` is a literal string `temperature_unit`

`self.hass.config.units.temperature_unit` is the units in `C` or `F`

Reply:
ah ok :+1: